CMakeLists.txt

```diff
@@ -435,7 +429,23 @@ if(NOT EXISTS ${LINKER_SCRIPT})
   message(FATAL_ERROR "Could not find linker script: '${LINKER_SCRIPT}'. Corrupted configuration?")
 endif()
 
-configure_file(version.h.in ${PROJECT_BINARY_DIR}/include/generated/version.h)
+if(DEFINED BUILD_VERSION)
+  set(build_version_argument "-DBUILD_VERSION=${BUILD_VERSION}")
+elseif(EXISTS ${ZEPHYR_BASE}/.git/index)
+  set(git_dependency ${ZEPHYR_BASE}/.git/index)
+else()
+  message(WARNING "ZEPHYR_BASE=${ZEPHYR_BASE} doesn't appear to be a git "
+    "repository, please specify '-DBUILD_VERSION=<version>'")
+endif()
+add_custom_command(
+  OUTPUT ${PROJECT_BINARY_DIR}/include/generated/version.h
+  COMMAND ${CMAKE_COMMAND} -DZEPHYR_BASE=${ZEPHYR_BASE}
+    -DOUT_FILE=${PROJECT_BINARY_DIR}/include/generated/version.h
+    ${build_version_argument}
+    -P ${ZEPHYR_BASE}/cmake/gen_version_h.cmake
```

Comment:
Did you forget to add this file to the PR ?

Reply:
i did.
Fixed.